backend/pkg/kubeconfig/watcher.go

```diff
@@ -106,3 +106,46 @@ func addFilesToWatcher(watcher *fsnotify.Watcher, paths []string) {
 		}
 	}
 }
+
+// syncContexts synchronizes the contexts in the store with the ones in the kubeconfig files.
+func syncContexts(kubeConfigStore ContextStore, paths string, source int) error {
+	// First read all kubeconfig files to get new contexts
+	newContexts, _, err := LoadContextsFromMultipleFiles(paths, source)
+	if err != nil {
+		return fmt.Errorf("error reading kubeconfig files: %v", err)
+	}
+
+	// Get existing contexts from store
+	existingContexts, err := kubeConfigStore.GetContexts()
+	if err != nil {
+		return fmt.Errorf("error getting existing contexts: %v", err)
+	}
+
+	// Find and remove contexts that no longer exist in the kubeconfig
+	for _, existingCtx := range existingContexts {
+		found := false
+
+		for _, newCtx := range newContexts {
+			if existingCtx.Name == newCtx.Name {
+				found = true
+
+				break
+			}
+		}
+
+		if !found {
+			err := kubeConfigStore.RemoveContext(existingCtx.Name)
+			if err != nil {
```

Comment:
This solution assumes that all contexts are loaded from the file that changed now only. There can be contexts from multiple files and different sources.

Reply:
I think even if they are from multiple store they would still be present in `kubeConfigStore` right? Line `119` does exactly the same. Do you think that even if there are other sources, they won't be present in the store?

Reply:
119 is listing all the contexts. If you see the code below there are three different types of sources from which we load the clusters from. Here you are listing(line 119) and removing all the contexts(line 137). We should remove only the clusters from kubeconfig source because that is what is triggering the sync here and they are only loaded in line 145.

https://github.com/headlamp-k8s/headlamp/blob/8fb46f63c2a51344478be29656ed36c928962f6d/backend/pkg/kubeconfig/kubeconfig.go#L28-L32

Reply:
Thank you for the explanation, updated it.